src/pages/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -37,66 +38,77 @@ class ServiceMetrics extends React.Component<ServiceId, ServiceMetricsState> {
     this.state = {
       loading: false
     };
+    this.renderChart = this.renderChart.bind(this);
   }
 
   componentDidMount() {
     this.getGrafanaInfo();
   }
 
-  onOptionsChanged = (options: MetricsOptions) => this.fetchMetrics(options);
+  onOptionsChanged = (options: MetricsOptions) => {
+    if (this.state.pollMetrics !== undefined) {
+      clearTimeout(this.state.pollMetrics);
+    }
+    this.fetchMetrics(options, this);
+  };
 
-  fetchMetrics(options: MetricsOptions) {
+  fetchMetrics(options: MetricsOptions, object: any) {
```

Comment:
Was the switch from `this` to `object` necessary? We could bind fetchMetrics to `this` in constructor, if it was the problem. (Or maybe writing it as arrow function would do the trick)

Reply:
Ok I fix this.

Reply:
thanks :)